### PR TITLE
Javascript unit test for validation

### PR DESCRIPTION
**Pull request by fatimarafiqui**
This PR includes the following:
- Separated validation method from circle_of_trust.js and index.js.
- Created validation.js
- Separation of phpunit tests and js unit tests in separate directories 
- Added unit tests for validation method in validationSpec.js

selection_001

@Anjali05 @MehaKaushik

### CODE REVIEW

**Comment by fatimarafiqui** [comment]
@Anjali05 @MehaKaushik Please review the PR.

**Comment by fatimarafiqui** [comment]
@Anjali05 @MehaKaushik Kindly review this.

**Review by Anjali05 — APPROVED** [review]

**Review by Anjali05 — COMMENTED** [review]
.

**Review by Anjali05 — APPROVED** [review]

File: tests/jsTests/validationSpec.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+describe('validateNumber', function() {
+    it('Should be phone number', function() {
+        expect(validation.isPhoneNumber('342345235')).toBeTruthy();
```
Comment (Anjali05) [inline tests/jsTests/validationSpec.js:7]: Is there any fixed length for the phone number or you just allow any length as long as it contains digits? if you are checking the length, include that as well. I am approving it now, you can add that in the next PR.

**Review by Anjali05 — APPROVED** [review]

**Review by Anjali05 — APPROVED** [review]

**Review by Anjali05 — APPROVED** [review]

**Review by Anjali05 — APPROVED** [review]

**Review by Anjali05 — DISMISSED** [review]

File: tests/jsTests/validationSpec.js (hunk 1 above)
Comment (fatimarafiqui) [reply tests/jsTests/validationSpec.js:7]: @Anjali05 sure.

File: js/validation.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const validation = {
+    isPhoneNumber:function(str) {
+        const pattern = /^\d+$/;
```
Comment (MehaKaushik) [inline js/validation.js:3]: We can use a better regex to check phone numbers. 
1. We can trim white spaces
2. Include country codes followed by hyphen
refer https://stackoverflow.com/questions/16699007/regular-expression-to-match-standard-10-digit-phone-number

Comment (MehaKaushik) [reply js/validation.js:3]: Once you modify this, add more negative tests for this code block.

File: tests/jsTests/validationSpec.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * Suite : validateEmail
+ * Description : Test suite to validate users' email address
+ */
+describe('validateEmail', function() {
+    it('Should be email', function() {
+        expect(validation.isEmailAddress('abc@def.com')).toBeTruthy();
+    });
+
+	it('Should be email', function() {
+        expect(validation.isEmailAddress('mail.google.com')).not.toBeTruthy();
```
Comment (MehaKaushik) [inline tests/jsTests/validationSpec.js:25]: Write some more negative tests. 
example : email@111.222.333.44444 this one is invalid
email@123.123.123.123 this one is valid 
email@-domain.com not valid
think of some edge cases if any

**Review by MehaKaushik — CHANGES_REQUESTED** [review]

File: tests/jsTests/validationSpec.js (hunk 3 above)
Comment (fatimarafiqui) [reply tests/jsTests/validationSpec.js:25]: @MehaKaushik This was just a basic test as an expectation for learning JS testing. The expectation was to add one test for any one method for any on the files to illustrate learning. As suggested by @Anjali05, More tests will be added at a later stage, I am still adding more dependencies and jasmine jquery libraries for testing.

Comment (MehaKaushik) [reply tests/jsTests/validationSpec.js:25]: I expect, that this would form a part of the module later. Why do you want to do things twice, if you finish this completely then you can directly plugin, this in the tests. 

Even otherwise I do not have any issues, we can go ahead, but while the actual testing we would have to redo this.

**Review by MehaKaushik — DISMISSED** [review]

File: tests/jsTests/validationSpec.js (hunk 3 above)
Comment (fatimarafiqui) [reply tests/jsTests/validationSpec.js:25]: @MehaKaushik Which module? There is not module/class involvement here.

**Review by MehaKaushik — APPROVED** [review]

File: tests/jsTests/validationSpec.js (hunk 3 above)
Comment (MehaKaushik) [reply tests/jsTests/validationSpec.js:25]: By module I meant the tests you would be writing for this file.

Comment (MehaKaushik) [reply tests/jsTests/validationSpec.js:25]: I am approving the PR, but these need to be incorporated in future PR when you are writing the tests for this particular file.
